cgroupstats.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2017-present, Outlyer
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+"""Cgroupstats module
+
+This module exists to expose the cgroupstats api to python
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# @lint-avoid-python-3-compatibility-imports
+# from __future__ import unicode_literals
+
+import struct
+import gnlpy.netlink as netlink
+
+
+class LoadStats(object):
+    __fields__ = [
+        'nr_sleeping', 'nr_running', 'nr_stopped',
+        'nr_uninterruptible', 'nr_iowait'
+    ]
+
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+    def __repr__(self):
+        arr = ['%s=%s' % (f, repr(self.__dict__[f])) for f in self.__fields__]
+        return 'LoadStats(%s)' % ', '.join(arr)
+
+    @staticmethod
+    def unpack(val):
+        fmt = 'QQQQQ'  # as per __fields__ above, and they're uint64 each
+        val = val[4:]  # strip attribute length, first 4 bytes
+        attrs = dict(zip(LoadStats.__fields__, struct.unpack(fmt, val)))
+        return LoadStats(**attrs)
+
+
+CgroupstatsCmdAttrList = netlink.create_attr_list_type(
+    'CgroupstatsAttrList',
+    ('CGROUPSTATS_CMD_ATTR_FD', netlink.U32Type),
+)
+
+
+CgroupstatsMessage = netlink.create_genl_message_type(
+    'CgroupstatsMessage', 'TASKSTATS',
+    ('NULL', netlink.IgnoreType),  # Reserved
+    ('GET', netlink.IgnoreType),
+    ('NEW', netlink.IgnoreType),
```

Comment:
@chantra What I've done is a bit of a hack, and it works because the `CGROUPSTATS_CMD_MAX` field ends up being equivalent to `4`; `STATS` is equivalent to `5`. This is what the kernel expects (in terms of integers), because as you can see here http://lxr.free-electrons.com/source/include/uapi/linux/cgroupstats.h the enum is initialised to `__TASKSTATS_CMD_MAX`, which is `3` (seen here: http://lxr.free-electrons.com/source/include/uapi/linux/taskstats.h).

So in order to only have `GET` and `NEW` (which I agree that we should do here), and therefore have them be equivalent to `4` and `5`, I think we need to change netlink.py (if I'm not mistaken). Basically the issue is that here https://github.com/facebook/gnlpy/blob/master/netlink.py#L301 the key is is taken from the attribute's position in the fields, and there isn't a way to initialise the key to a value.

As for the complex type, as far as I could see the kernel emits an attribute with the following structure:
[AttributeLength (4 bytes), NrSleeping (8 bytes), NrRunning (8 bytes), NrStopped (8 bytes), NrUninterruptible (8 bytes), NrIoWait (8 bytes)]

So if I define it as an attribute list type of 5 different fields of `netlink.U64Type` type each, the issue I have is that the code expects the kernel to send 5 different structures of [AttributeLength, Attribute], but from what I could see that doesn't match with what the kernel actually sends.

I could define it as one single attribute, with a custom type that would correspond to 40 bytes (5 * 8 bytes). But I would then still need to decapsulate the 5 different items onto some structure so they could be accessed separately.

Let me know if you want me to paste some example data that the kernel sends!